projects/client/RabbitMQ.Client/src/client/api/IConnectionFactory.cs

```diff
@@ -129,13 +129,6 @@ public interface IConnectionFactory
         IConnection CreateConnection(IList<string> hostnames, String clientProvidedName);
 
         /// <summary>
-        /// Advanced option.
-        ///
-        /// What task scheduler should consumer dispatcher use.
-        /// </summary>
-        TaskScheduler TaskScheduler { get; set; }
```

Comment:
I'm fine with restoring it if that will make this `4.1.0` according to the definition of semantic version you provided. In my opinion, we're in this grayish area of the semantic versioning. 
I'm pushing the commit right away.